store/rootmulti/store.go

```diff
@@ -596,20 +607,12 @@ type storeCore struct {
 	// ... maybe add more state
 }
 
-// Implements merkle.Hasher.
+// Hash is the value we commit to in SimpleHashFromMap
+// Which should equal the root hash of the substore
 func (si storeInfo) Hash() []byte {
 	// Doesn't write Name, since SimpleHashFromMap() will
 	// include them via the keys.
-	bz := si.Core.CommitID.Hash
```

Comment:
The `storeInfo` contains a name and a commitment (hash and version).
This method is not really used to "hash" it in a normal means, but rather pull out the hash that was already inside of it.

In effect, to make a chained proof op work, you would have to take the root hash from the iavlstore and then hash it before feeding it as a value to the merkle simple proof, which makes no sense. We want the simple proof to commit directly to the root hash of the sub stores (not a hash of a hash of a hash... - we are not bitcoin mining).

If anyone intends to use this as a real hashing op (which it never was used as), it must contain the name as well before feeding it to the hasher. In such a case we could rename this to eg. `Commitment`, which we would use to feed into `commitInfo.Hash()` and `commitInfo.ProofOp()`.

Reply:
Yes, believe this was why the hashing in TM had to be removed. This approach works better. Could you please put the cruz of this comment in the godoc. Namely that this does not actually hash the storeInfo, rather it returns the embedded store hash

Reply:
Yes, I can definitely add it to the GoDoc. I can also rename the function to make it more explicit that we are not trying to implement the `Hasher` interface but just access the substore hash.


Actually I can do both (rename an explain the purpose of the function). Any preferences for names?